nixos/modules/services/security/tang.nix

```diff
@@ -0,0 +1,95 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.tang;
+in
+{
+  options.services.tang = {
+    enable = mkEnableOption "tang";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.tang;
+      defaultText = literalExpression "pkgs.tang";
+      description = mdDoc "The tang package to use.";
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 7654;
+      description = mdDoc "The port to listen on.";
+    };
+
+    whitelist = mkOption {
+      default = [ ];
+      example = [ "192.168.1.0/24" ];
+      type = types.listOf types.str;
+      description = ''
+        Whitelist a list of address prefixes.
+        Preferably, internal addresses should be used.
+      '';
+    };
+
+  };
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ pkgs.tang ];
+
+    systemd.services."tangd@" = {
+      description = "Tang server";
+      path = [ pkgs.tang ];
+      serviceConfig = {
+        StandardInput = "socket";
+        StandardOutput = "socket";
+        StandardError = "journal";
+        User = "tang";
+        Group = "tang";
+        DynamicUser = true;
+        StateDirectory = "tang";
+        RuntimeDirectory = "tang";
+        StateDirectoryMode = "700";
+        UMask = "0077";
+        CapabilityBoundingSet = [ "" ];
+        ExecStart = "${cfg.package}/libexec/tangd $STATE_DIRECTORY";
```

Comment:
I think the syntax for using an environment variable in `ExecStart` is `${STATE_DIRECTORY}`, not `$STATE_DIRECTORY`, isn't it? Regardless, `%S` always expands to the state directory, so it's probably best to use that.

Reply:
:+1: